lib/omniauth/strategies/openid_connect.rb

```diff
@@ -109,7 +109,8 @@ def callback_phase
 
         raise CallbackError, 'Invalid state parameter' if invalid_state
 
-        return fail!(:missing_code, OmniAuth::OpenIDConnect::MissingCodeError.new(params['error'])) unless params['code']
+        return fail!(:missing_code, OmniAuth::OpenIDConnect::MissingCodeError.new(params['error'])) unless valid_response_type?('code')
```

Comment:
Please consider extracting these two lines into separate `#validate` method and rewrite current validation method `#valid_response_type?(type)` to make it more readable:

```ruby
return fail!(:missing_code, OmniAuth::OpenIDConnect::Missing...
return fail!(:missing_code, OmniAuth::OpenIDConnect::Missing...
```

⬇️ 

```ruby
return unless valid_request_type?
```

and

```ruby
def valid_response_type?(type)
  # pass if configured response_type is different than tested here
  options.response_type == type.to_s ? params.key?(type) : true
end
```

⬇️ 

```ruby
TYPES_MAP = {
  'id_token' => { exception_class: OmniAuth::OpenIDConnect::MissingIdTokenError, key: :missing_id_token }.freeze,
  'code' => { exception_class: OmniAuth::OpenIDConnect::MissingCodeError, key: :missing_code }.freeze,
}.freeze

def valid_request_type?
  return true if params.key?(options.response_type)

  # TODO: implement fallback exception
  fallback_exception = nil
  error_attrs = TYPES_MAP.fetch(options.response_type, fallback_exception)
  fail!(error_attrs[:key], error_attrs[:exception_class].new(params['error']))

  false
end
```

Reply:
Makes sense, it's updated.